hwloc/topology.c

```diff
@@ -1586,7 +1613,12 @@ propagate_unused_cpuset(hwloc_obj_t obj, hwloc_obj_t sys)
 	hwloc_bitmap_and(obj->complete_cpuset, obj->complete_cpuset, sys->complete_cpuset);
       } else {
 	obj->complete_cpuset = hwloc_bitmap_dup(sys->complete_cpuset);
-	hwloc_bitmap_and(obj->complete_cpuset, obj->complete_cpuset, obj->cpuset);
+        if (obj->complete_cpuset)
+	  hwloc_bitmap_and(obj->complete_cpuset, obj->complete_cpuset, obj->cpuset);
+        else {
+          hwloc_bitmap_free(mask);
+          return -1;
+        }
       }
```

Comment:
All this core code is under total rework in a branch. And there are flaws in the bitmap API anyway because it can realloc internally without reporting errors. We'll come back to all these cpuset stuff in a couple months, you can drop it for now.